presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -300,6 +301,15 @@ private static int hashBytes(int initialValue, Slice bytes)
             return Optional.of(new HiveBucketFilter(ImmutableSet.of(singleBucket.getAsInt())));
         }
 
+        Optional<Map<ColumnHandle, Set<NullableValue>>> columnValueSetBindings = TupleDomain.extractFixedValuesFromRanges(effectivePredicate);
+        if (!columnValueSetBindings.isPresent()) {
+            return Optional.empty();
+        }
+        Set<Integer> buckets = getHiveBuckets(table, columnValueSetBindings.get());
+        if (buckets.size() > 0) {
```

Comment:
@gparai What if there is no match? The desired behavoir would be to not read any file. Is this how it would work?

Reply:
We would always get some bucket for the given binding unless something was not right/valid in which case it would be `Optional.empty()`. I have removed this check and rely on the optional instead.

Reply:
@gparai You are right. Thanks for clarifying.